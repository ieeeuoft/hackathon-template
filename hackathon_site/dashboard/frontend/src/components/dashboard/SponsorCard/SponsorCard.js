import React from "react";
import styles from "./SponsorCard.module.scss";
import Slider from "react-slick";
import Typography from "@material-ui/core/Typography";
import Paper from "@material-ui/core/Paper";
import Grid from "@material-ui/core/Grid";
import { sponsorsList } from "testing/mockData";

let settings = {
    dots: true,
    arrows: true,
    autoplaySpeed: 2500,
    autoplay: true,
    slidesToShow: 2,
    speed: 500,
    pauseOnHover: false,
};

export const UnconnectedSponsorCard = ({ sponsors }) => {
    return !sponsors.length ? null : (
        <Grid className={styles.sponsors} data-testid="sponsor-card" item>
            <Typography variant="h2">Thanks to our sponsors!</Typography>
            <Paper elevation={3} className={styles.sponsorsPaper} square={true}>
                <Slider {...settings}>
                    {sponsors.map((item, i) => (
                        <img
                            src={require("assets/images/sponsors/" + item.imgSrc)}
                            alt={item.imgSrc}
                            key={i}
                        />
                    ))}
                </Slider>
            </Paper>
        </Grid>
    );
};

const ConnectedSponsorCard = () => <UnconnectedSponsorCard sponsors={sponsorsList} />;

export default ConnectedSponsorCard;
