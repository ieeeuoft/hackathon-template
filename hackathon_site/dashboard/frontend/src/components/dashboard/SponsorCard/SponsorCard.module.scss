@import "./../../../assets/abstracts/mixins";
@import "./../../../assets/abstracts/variables";

.sponsors {
    height: 100%;
    width: 400px;
    @include flexPosition($dir: column);
    &Paper {
        flex-grow: 1;
        padding: 28px 0;
        img {
            height: 100%;
        }
    }
}
