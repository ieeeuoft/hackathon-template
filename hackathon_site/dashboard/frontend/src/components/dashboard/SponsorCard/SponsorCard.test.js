import React from "react";
import { render } from "@testing-library/react";
import { UnconnectedSponsorCard } from "./SponsorCard";

it("renders correctly when there are 2 images", () => {
    let testList = [{ imgSrc: "AMD.svg" }, { imgSrc: "ECE.png" }];
    const { queryByAltText } = render(<UnconnectedSponsorCard sponsors={testList} />);
    for (let e of testList) {
        expect(queryByAltText(e.imgSrc)).toBeTruthy();
    }
});
