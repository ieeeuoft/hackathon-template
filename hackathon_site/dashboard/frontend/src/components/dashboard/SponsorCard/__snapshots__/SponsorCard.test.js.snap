// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are 2 images 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root sponsors MuiGrid-item"
    data-testid="sponsor-card"
  >
    <h2
      class="MuiTypography-root MuiTypography-h2"
    >
      Thanks to our sponsors!
    </h2>
    <div
      class="MuiPaper-root sponsorsPaper MuiPaper-elevation3 MuiPaper-rounded"
    >
      <div
        class="slick-slider slick-initialized"
      >
        <div
          class="slick-list"
        >
          <div
            class="slick-track"
            style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
          >
            <div
              aria-hidden="false"
              class="slick-slide slick-active slick-current"
              data-index="0"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <img
                  alt="AMD.svg"
                  src="[object Object]"
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                />
              </div>
            </div>
            <div
              aria-hidden="false"
              class="slick-slide slick-active"
              data-index="1"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <img
                  alt="ECE.png"
                  src="ECE.png"
                  style="width: 100%; display: inline-block;"
                  tabindex="-1"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
